Cannot add multiple out-of-the-tree secure partitions

Zephyr is not handling correctly the parameter -DTFM_EXTRA_PARTITION_PATHS when it contains multiple paths. According to documentation, path should be inside double quote and separated by ;.
```
-DTFM_EXTRA_MANIFEST_LIST_FILES="<Absolute-path-part-1-folder/manifes_list.yaml>;<Absolute-path-part-2-folder/manifes_list.yaml>"
-DTFM_EXTRA_PARTITION_PATHS="<Absolute-path-part-1-folder>;<Absolute-path-part-2-folder>"
```
When I try to use it in my code with:
```
set_property(TARGET zephyr_property_target
  APPEND PROPERTY TFM_CMAKE_OPTIONS
  ...
  -DTFM_EXTRA_PARTITION_PATHS="${MY_PARTITION_DIR};${DUMMY_PARTITION_DIR}"
)
```
I get the error because two paths are treated as one path, with space inside. 
```
CMake Error at secure_fw/partitions/CMakeLists.txt:58 (add_subdirectory):
  add_subdirectory given source
  "C:/Users/kasia/zephyrproject/zephyr/zephyr-integration-tests/my_partition
  C:/Users/kasia/zephyrproject/zephyr/zephyr-integration-tests/tfm_secure_partition/dummy_partition"
  which is not an existing directory.
```
In CMakeCache.txt I also see the paths are separated by space:
```
TFM_EXTRA_PARTITION_PATHS:PATH=C:/Users/kasia/zephyrproject/zephyr/zephyr-integration-tests/my_partition C:/Users/kasia/zephyrproject/zephyr/zephyr-integration-tests/tfm_secure_partition/dummy_partition
```
Same issue happens to multiple manifest file.

--- a/doc/services/tfm/build.rst
+++ b/doc/services/tfm/build.rst
@@ -122,6 +122,18 @@ following CMake snippet in your CMakeLists.txt file.
    options. Also CMake generator expressions are supported, such as
    ``$<1:-DFOO=bar>``
 
+Since ``TFM_CMAKE_OPTIONS`` is a list argument it will be expanded before it is
+passed to the TF-M build system.
+Options that have list arguments must therefore be properly escaped to avoid
+being expanded as a list.
+
+   .. code-block:: cmake
+
+     set_property(TARGET zephyr_property_target
+                  APPEND PROPERTY TFM_CMAKE_OPTIONS
+                  -DFOO="bar\\\;baz"
+     )
+
 Footprint and Memory Usage
 **************************
 

